app/controllers/api/integrations_controller.rb

```diff
@@ -201,6 +203,17 @@ def authorize
     authorize! :edit, @service
   end
 
+  def update_mapping_rules_position
+    proxy_rules_attributes.each_value do |attrs|
+      proxy_rule = @proxy.proxy_rules.find_by(id: attrs['id'])
+      proxy_rule&.set_list_position(attrs['position'])
```

Comment:
Then remove the nil check

```suggestion
      proxy_rule.set_list_position(attrs['position'])
```